client/core/core.go

```diff
@@ -6557,3 +6559,203 @@ func (c *Core) WalletLogFilePath(assetID uint32) (string, error) {
 
 	return wallet.LogFilePath()
 }
+
+func (c *Core) DeleteArchivedRecords(olderThan *time.Time, matchesFile, ordersFile *os.File) error {
+	c.wg.Add(1)
+	c.wg.Done()
+	var (
+		finished  bool
+		err       error
+		csvWriter *csv.Writer
+	)
+
+	// If provided a file to write the orders csv to, write the header and
+	// defer closing the file.
+	if ordersFile != nil {
+		defer ordersFile.Close()
```

Comment:
I tend to think that the file should be closed by whatever opened it e.g. rpcserver, not here.

But I think `DeleteArchivedRecords` should take `io.Writer`s, not `*os.File`.  The caller could conceivably be the webserver and it would be desirable to write to an `http.ResponseWriter`...
although how that I say it, it's problematic that there are two different outputs, one for matches and one for orders.  Hmm.  Merging them would make for a nasty csv scheme.

Reply:
Can just pass the file name and not verify it in the rpcserver.

Reply:
Just passing the file name now. Core handles opening the file and clean up.